src/web/components/Callout/Form.tsx

```diff
@@ -0,0 +1,158 @@
+import { TextInput } from './TextInput';
+
+const formStyles = css`
+    display: flex;
+    flex-direction: column;
+    justify-content: space-between;
+    padding-left: 10px;
+    padding-right: 10px;
+`;
+
+const footerPaddingStyles = css`
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+`;
+
+type addFormFieldProp = {
+    formField: CampaignFieldType;
+    formData: { [key in string]: any };
+    setFormData: React.Dispatch<React.SetStateAction<{ [x: string]: any }>>;
+};
+
+const addFormField = ({
+    formField,
+    formData,
+    setFormData,
+}: addFormFieldProp) => {
+    switch (formField.type) {
+        case 'textarea':
+            return (
+                <TextArea
+                    formField={formField}
+                    formData={formData}
+                    setFormData={setFormData}
+                />
+            );
+        case 'file':
+            return (
+                <FileUpload
+                    formField={formField}
+                    formData={formData}
+                    setFormData={setFormData}
+                />
+            );
+        case 'select':
+            return (
+                <Select
+                    formField={formField}
+                    formData={formData}
+                    setFormData={setFormData}
+                />
+            );
+        case 'radio':
+            return (
+                <Radios
+                    formField={formField}
+                    formData={formData}
+                    setFormData={setFormData}
+                />
+            );
+        case 'checkbox':
+            return (
+                <Checkboxes
+                    formField={formField}
+                    formData={formData}
+                    setFormData={setFormData}
+                />
+            );
+        default: {
+            return (
+                <TextInput
+                    formField={formField}
+                    formData={formData}
+                    setFormData={setFormData}
+                />
+            );
+        }
+    }
+};
+
+type FormProps = {
+    submitForm: () => void;
+    formId: number;
+    formFields: CampaignFieldType[];
+    error?: string;
+};
+
+export const Form = ({ submitForm, formId, formFields, error }: FormProps) => {
+    const [twitterHandle, setTwitterHandle] = useState('');
+    const [formData, setFormData] = useState<{ [key in string]: any }>({});
+
+    return (
+        <form
+            action="/formstack-campaign/submit"
+            method="post"
+            className={formStyles}
+            onSubmit={e => {
+                e.preventDefault();
+                submitForm();
+            }}
+        >
+            <input name="formId" type="hidden" value={formId} />
+            {formFields.map((formField, index) => (
+                <div key={index} custom-guardian="callout-form-field">
+                    {addFormField({
```

Comment:
You don't need to use a function here. A better method would be to compose using a component. Eg:
```
<AddFormField prop={blah} />
```

And in this case the verb 'add' probably makes less sense and you could just name it `Field` or something